src/software/ai/navigator/obstacle/obstacle_factory.cpp

```diff
@@ -142,31 +147,62 @@ std::vector<ObstaclePtr> ObstacleFactory::createVelocityObstaclesFromTeam(
     return obstacles;
 }
 
-ObstaclePtr ObstacleFactory::createBallObstacle(const Point &ball_position)
+ObstaclePtr ObstacleFactory::createBallObstacle(const Point &ball_position) const
+{
+    return createInflatedByRobotRadius(Circle(ball_position, BALL_MAX_RADIUS_METERS));
+}
+
+ObstaclePtr ObstacleFactory::createRobotObstacle(const Point &robot_position) const
+{
+    return createInflatedByRobotRadius(Circle(robot_position, ROBOT_MAX_RADIUS_METERS));
+}
+
+ObstaclePtr ObstacleFactory::createObstacleFromRectangle(const Rectangle &rectangle) const
 {
-    return createObstacle(
-        Circle(ball_position, BALL_MAX_RADIUS_METERS + 0.06 + shape_expansion_amount));
+    return createInflatedByRobotRadius(Polygon(rectangle));
 }
 
-ObstaclePtr ObstacleFactory::createRobotObstacle(const Point &robot_position)
+ObstaclePtr ObstacleFactory::createInflatedByRobotRadius(const Circle &circle) const
```

Comment:
Just as a quick note, I know I suggested this name, looking at in the context of the rest of the class, IMO we should change `create` to `createObstacle` or change the other functions to be consistent with this one. I'm totally fine with either option.

Reply:
Here's another thought, all `Obstacle`s created by the `ObstacleFactory` implicitly include a robot radius margin around them, but for clarity I add a comment to explicitly state that where it's not intuitive.

Reply:
After offline discussion, ddditionally renamed ObstacleFactory to RobotNavigationObstacleFactory